timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -177,9 +179,11 @@ private AsyncResult<HeldLocks> acquireImmutableTimestampLock(UUID requestId, lon
     }
 
     private AsyncResult<HeldLocks> acquireNamedTimestampLockInternal(
-            TimestampLeaseName timestampName, UUID requestId, long timestamp) {
-        AsyncLock lock = lockManager.getNamedTimestampLock(timestampName, timestamp);
-        return lockAcquirer.acquireLocks(requestId, OrderedLocks.fromSingleLock(lock), TimeLimit.zero());
+            UUID requestId, Set<TimestampLeaseName> timestampNames, long timestamp) {
+        List<AsyncLock> locks = timestampNames.stream()
+                .map(name -> lockManager.getNamedTimestampLock(name, timestamp))
+                .collect(Collectors.toList());
+        return lockAcquirer.acquireLocks(requestId, OrderedLocks.fromOrderedList(locks), TimeLimit.zero());
```

Comment:
*I think* the ```OrderedLocks.fromOrderedList(locks)``` is correct, because the timestamp is unique, so you can't really have anyone competing and therefore you can't have livelock from people trying to grab locks in different orders.

BUT it reads funny for sure, maybe let's just order the initial set that would give you what this should be?

Reply:
Added comment to your comment, I think we can track this upstairs.